File: packsecure-flutter/lib/core/services/print_service.dart
import 'package:flutter/foundation.dart';

class LabelData {
  final String orderNo;
  final String skuName;
  final String? cuttingSize;
  final int rollsCount;
  final String operatorName;
  final String factoryName;

  LabelData({
    required this.orderNo,
    required this.skuName,
    this.cuttingSize,
    required this.rollsCount,
    required this.operatorName,
    required this.factoryName,
  });
}

/// 工业标签静默打印驱动
class GooglePrintService {
  /// 静默打印机台标签 (零弹窗直接出单)
  static Future<bool> printLabelSilent(LabelData data) async {
    if (kDebugMode) {
      print('[Flutter Print] 向串口/蓝牙热敏打印机发送 ESC/POS 指令...');
      print('[Flutter Print] 打印工单: ${data.orderNo}, SKU: ${data.skuName}, 数量: ${data.rollsCount} 卷');
    }

    // 模拟 250ms 静默秒级打印
    await Future.delayed(const Duration(milliseconds: 250));
    return true;
  }
}

File: packsecure-mobile/src/services/printService.ts
/**
 * Packsecure 工厂机台静默标签打印驱动
 * 支持蓝牙热敏打印机、网络 TCP 标签打印机静默出单
 */
export interface PrintLabelData {
  orderNo: string;
  skuName: string;
  cuttingSize?: string;
  rollsCount: number;
  operatorName: string;
  factoryName: string;
  timestamp: string;
}

export const PrintService = {
  /**
   * 搜索附近的蓝牙标签打印机
   */
  scanBluetoothPrinters: async () => {
    console.log('[Print Engine] 开始搜索车间蓝牙热敏打印机...');
    // 返回模拟检测到的蓝牙打印机列表
    return [
      { id: 'BT-PRINTER-01', name: 'Factory Label Printer T1 (Bluetooth)', address: '00:11:22:33:44:55' },
      { id: 'BT-PRINTER-02', name: 'Mobile Thermal Printer - Driver (Bluetooth)', address: '66:77:88:99:AA:BB' }
    ];
  },

  /**
   * 静默打印机台标签 (扫码后零弹窗瞬间出单)
   */
  printMachineLabelSilent: async (printerAddress: string, data: PrintLabelData): Promise<{ success: boolean; message: string }> => {
    console.log(`[Print Engine] 向打印机设备 [${printerAddress}] 发送 ESC/POS 标签指令...`);
    console.log(`[Print Engine] 打印内容: 单号=${data.orderNo}, SKU=${data.skuName}, 数量=${data.rollsCount}卷, 操作员=${data.operatorName}`);

    // 构建 ESC/POS 二进制打单指令 (模拟发送串口/蓝牙字节流)
    try {
      // 模拟 300ms 完成硬打印
      await new Promise((resolve) => setTimeout(resolve, 300));
      return { success: true, message: '标签静默打印成功！' };
    } catch (err: any) {
      return { success: false, message: `打印机响应超时或出纸异常: ${err.message}` };
    }
  },

  /**
   * 静默打印司机送货单据
   */
  printDeliveryReceiptSilent: async (printerAddress: string, tripNo: string, customerName: string): Promise<boolean> => {
    console.log(`[Print Engine] 打印司机送货单据: TripNo=${tripNo}, 客户=${customerName}`);
    await new Promise((resolve) => setTimeout(resolve, 400));
    return true;
  }
};
